internal/scans/scan_progress.go

```diff
@@ -0,0 +1,67 @@
+package scans
+
+import (
+	"context"
+	"time"
+
+	"github.com/rs/zerolog/log"
+)
+
+const timeout = 5 * time.Second
+
+// ScanProgress is a type that's used for tracking current running scans, listen to their progress events
+// and invoke cancellations or "done" signals. This allows throttling or cancelling previous scans instead of
+// having many scans running at once for the same files.
+type ScanProgress struct {
+	isDone bool
+	done   chan struct{}
+	cancel chan struct{}
+}
+
+func NewScanProgress() *ScanProgress {
+	return &ScanProgress{
+		cancel: make(chan struct{}),
+		done:   make(chan struct{}),
+	}
+}
+
+func (rs *ScanProgress) GetDoneChannel() <-chan struct{}   { return rs.done }
```

Comment:
why not a typed channel?

Reply:
I don't have any data to pass so it's essentially a signal-only channel.

I can switch to `chan bool` and write `true` if it's less confusing/ugly

Reply:
yeah, I'd prefer `bool` - easier on the eyes and to understand I think.